### PR TITLE
Rw join ou

### PR DESCRIPTION
**Pull request by cptera**

### CODE REVIEW

File: azuredeploy.json [hunk 1]
```diff
@@ -163,7 +163,7 @@
       "metadata": {
         "description": "The location of resources, such as templates and DSC modules, that the template depends on"
       },
-      "defaultValue": "https://raw.githubusercontent.com/teradici/deploy/master"
+      "defaultValue": "https://raw.githubusercontent.com/teradici/deploy/rw-join-ou"
```
Comment (cptera) [inline azuredeploy.json]: This needs to be reverted before merging

File: azuredeploy.parameters.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
             "value": "https://teradeploy.blob.core.windows.net/binaries"
         },
         "_artifactsLocation": {
-            "value": "https://raw.githubusercontent.com/teradici/deploy/master"
+            "value": "https://raw.githubusercontent.com/teradici/deploy/rw-join-ou"
```
Comment (cptera) [inline azuredeploy.parameters.json]: This needs to be reverted before merging

File: remote-workstations/new-agent-vm/rhel-standard-agent.json [hunk 3]
```diff
@@ -69,6 +69,13 @@
                 "description": "Set of bit flags that define the join options. Default value of 3 is a combination of NETSETUP_JOIN_DOMAIN (0x00000001) & NETSETUP_ACCT_CREATE (0x00000002) i.e. will join the domain and create the account on the domain. For more information see https://msdn.microsoft.com/en-us/library/aa392154(v=vs.85).aspx"
             }
         },
+        "domainOrganizationUnitToJoin": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "OU for VM to join to"
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/rhel-standard-agent.json]: I think we can lose the second 'to'

File: remote-workstations/new-agent-vm/server2016-graphics-agent.json [hunk 4]
```diff
@@ -80,37 +88,10 @@
...
-            "metadata": {
-                "description": "URI of the user's storage account"
-            }
-        },
-        "userStorageAccountSasToken": {
-            "type": "securestring",
-            "metadata": {
-                "description": "SAS token for accessing user's storage account"
-            },
-            "defaultValue": "?public"
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/server2016-graphics-agent.json]: ?public wasn't needed?

File: remote-workstations/new-agent-vm/server2016-standard-agent.json [hunk 5]
```diff
@@ -68,6 +69,13 @@
                 "description": "Set of bit flags that define the join options. Default value of 3 is a combination of NETSETUP_JOIN_DOMAIN (0x00000001) & NETSETUP_ACCT_CREATE (0x00000002) i.e. will join the domain and create the account on the domain. For more information see https://msdn.microsoft.com/en-us/library/aa392154(v=vs.85).aspx"
             }
         },
+        "domainOrganizationUnitToJoin": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "OU for VM to join to"
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/server2016-standard-agent.json]: two to's

File: remote-workstations/new-agent-vm/server2016-graphics-agent.json (hunk 4 above)
Comment (cptera) [reply remote-workstations/new-agent-vm/server2016-graphics-agent.json]: Apparently it was a workaround for an issue that's been fixed

File: remote-workstations/new-agent-vm/server2016-graphics-agent.json [hunk 6]
```diff
@@ -68,6 +69,13 @@
                 "description": "Set of bit flags that define the join options. Default value of 3 is a combination of NETSETUP_JOIN_DOMAIN (0x00000001) & NETSETUP_ACCT_CREATE (0x00000002) i.e. will join the domain and create the account on the domain. For more information see https://msdn.microsoft.com/en-us/library/aa392154(v=vs.85).aspx"
             }
         },
+        "domainOrganizationUnitToJoin": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "OU for VM to join to"
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/server2016-graphics-agent.json]: two to's

File: remote-workstations/new-agent-vm/rhel-standard-agent.json [hunk 7]
```diff
@@ -149,11 +155,8 @@
         }
     },
     "variables": {
-
         "apiVersion": "2015-06-15",
         "nicName": "[concat(parameters('dnsLabelPrefix'),'Nic')]",
-        "enableAutoShutdown" : false,
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/rhel-standard-agent.json]: yeah this is always enabled on Linux and isn't controlled from here?

File: remote-workstations/new-agent-vm/Install-PCoIPAgent.sh [hunk 8]
```diff
@@ -102,7 +104,19 @@ sudo yum -y install sssd realmd oddjob oddjob-mkhomedir adcli samba-common samba
 sudo systemctl enable sssd
 
 echo "-->Joining the domain"
-echo "$PASSWORD" | sudo realm join --user="$USERNAME" "$DOMAIN_NAME"
+if [ ${OU} ]
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/Install-PCoIPAgent.sh]: Curiosity - so when should we use if [ ${OU} ] versus if [ -n ${OU} ] or should it be if [ -n "${OU}" ] ? (I have a leaning for the latter based on some examples I saw but I don't really know why.)

File: remote-workstations/new-agent-vm/Install-PCoIPAgent.sh [hunk 9]
```diff
@@ -74,6 +74,8 @@ STORAGEURI="$9"
 COLLECTOR_ID="${10}"
 # the eleventh argument is the SaS Storage account token
 SAS_TOKEN="${11}"
+# the twelfth arguement is the Computer OU string
```
Comment (bill-dall) [inline remote-workstations/new-agent-vm/Install-PCoIPAgent.sh]: arguement -> argument

File: remote-workstations/new-agent-vm/rhel-standard-agent.json (hunk 7 above)
Comment (cptera) [reply remote-workstations/new-agent-vm/rhel-standard-agent.json]: this didn't get used anywhere

File: remote-workstations/new-agent-vm/Install-PCoIPAgent.sh (hunk 8 above)
Comment (cptera) [reply remote-workstations/new-agent-vm/Install-PCoIPAgent.sh]: yeah, I think the latter is safer. mine might break if it's an int or something unexpected like that

**Pull request merged**